doc/mgmt/gnmi/SONiC_GNMI_Server_Interface_Design.md

```diff
@@ -441,18 +441,42 @@ For incremental configuration, 'config apply-patch' should not restart gNMI, bgp
 
 ##### 1.2.1.11.2 Full Configurations
 
-For full configuration, there’re 2 possible solutions: 
-* gNMI server needs to send response at first, and then invoke 'config reload'.
-
-<img src="images/full_rpc.svg" alt="full-gnmi" width="800px"/>
-
-* Use gNMI request and gNOI request together to implement full configuration update.
-
-<img src="images/full_rpc_gnoi.svg" alt="full-gnmi-gnoi" width="800px"/>
-
-The full configuration request will be overwritten by subsequent full configuration request or incremental configuration request.
-
-<img src="images/mixed requests.svg" alt="overwritten-config" width="800px"/>
+We will use single gnmi request to support full configuration update.
+1. The host service will perform YANG validation and return an error if it fails.
+2. The host service will mask the gNMI service, preventing it from restarting after a config reload.
+3. The host service will execute a config reload using the input configuration.
```

Comment:
is there any chance that switch will not be in desired state after config reload, if so will it be handled here? like rollback?

Reply:
yes, we will use "config reload -f" to rollback